src/ParallelAlgorithms/LinearSolver/Gmres/InitializeElement.hpp

```diff
@@ -46,8 +46,22 @@ struct InitializeElement {
           Convergence::Tags::IterationId<OptionsGroup>>;
   using basis_history_tag =
       LinearSolver::Tags::KrylovSubspaceBasis<operand_tag>;
+  using preconditioned_basis_history_tag =
+      LinearSolver::Tags::KrylovSubspaceBasis<preconditioned_operand_tag>;
 
  public:
+  using simple_tags = tmpl::append<
+      tmpl::list<Convergence::Tags::IterationId<OptionsGroup>,
+                 initial_fields_tag, operator_applied_to_fields_tag,
+                 operand_tag, operator_applied_to_operand_tag,
+                 orthogonalization_iteration_id_tag, basis_history_tag,
+                 Convergence::Tags::HasConverged<OptionsGroup>>,
+      tmpl::conditional_t<Preconditioned,
+                          tmpl::list<preconditioned_basis_history_tag,
+                                     preconditioned_operand_tag>,
+                          tmpl::list<>>>;
```

Comment:
I remember `tmpl::append` had some bug related to empty lists, @kidder @wthrowe @nilsdeppe can you remind me what the failure mode actually is? I have been using this pattern recently:

```suggestion
  using simple_tags = tmpl::flatten<tmpl::list<
      Convergence::Tags::IterationId<OptionsGroup>,
                 initial_fields_tag, operator_applied_to_fields_tag,
                 operand_tag, operator_applied_to_operand_tag,
                 orthogonalization_iteration_id_tag, basis_history_tag,
                 Convergence::Tags::HasConverged<OptionsGroup>,
      tmpl::conditional_t<Preconditioned,
                          tmpl::list<preconditioned_basis_history_tag,
                                     preconditioned_operand_tag>,
                          tmpl::list<>>>>;
```

Reply:
The container has a new enough version of Brigand that that bug is fixed, so if it passes CI it's fine.  (See also all the failures on #2326.)

Reply:
ok then I'm happy with the code as-is